pkg/cmd/namespace.go

```diff
@@ -70,9 +81,18 @@ func runCostNamespace(ko *KubeOptions, no *CostOptionsNamespace) error {
 			currencyCode,
 		)
 	} else {
-		allocations, err := query.QueryAllocation(ko.clientset, *ko.configFlags.Namespace, no.serviceName, no.window, "namespace", context.Background())
-		if err != nil {
-			return fmt.Errorf("failed to query allocation API: %s", err)
+		var allocations []map[string]kubecost.Allocation
+		var err error
+		if no.useProxy {
+			allocations, err = query.QueryAllocation(ko.clientset, *ko.configFlags.Namespace, no.serviceName, no.window, "namespace", context.Background())
+			if err != nil {
+				return fmt.Errorf("failed to query allocation API: %s", err)
+			}
+		} else {
+			allocations, err = query.QueryAllocationFwd(ko.restConfig, *ko.configFlags.Namespace, no.serviceName, no.window, "namespace", context.Background())
+			if err != nil {
+				return fmt.Errorf("failed to query allocation API: %s", err)
+			}
```

Comment:
Did you consider a design that would allow for a single `QueryAllocation` function? Like, perhaps setting `useProxy` as a value in the `context` and making the decision internally to the function? I wouldn't say I feel strongly about this, but having `n` different entire versions of the API based on configuration flags feels potentially cumbersome.

Reply:
It's a good thought. The only possible future extension I see to query paradigms is a custom URL that Kubecost has been exposed at. That's really very different behavior and requires a different set of parameters, so I wouldn't put it under the same function. When that day comes, the most important duplicated data is imo the construction of the query parameters within `QueryAllocation(Fwd)`. I'd say those can get refactored into a single initialization then.

I'm really tempted to therefore just put a boolean flag on `QueryAllocation` and `QueryAggCostModel` for `useProxy`, but I'm trying to adhere to a "no boolean behavior flags on exported functions" thing so that's out.

It's a little boilerplate-y right now, but I don't mind it too much. I think there's a lot of potential for refactoring and generating a lot of these subcommands automatically instead of the intense duplication going on right now. Something like "generate a subcommand for the allocation API with a `pod` aggregation" instead of making a new file `pod.go` and just changing some names. Well, anyway that's neither here nor there. When I get around to adding assets subcommands I think that refactor will happen.

Reply:
Well, I changed my mind in https://github.com/kubecost/kubectl-cost/pull/82